### ISSUE TITLE
grails-logging doesn't honor dynamic log level changes

### ISSUE THREAD

**Issue by jfrosch** [grails/grails-core#11000]
### Expected Behaviour

The Spring Boot **/loggers** endpoint accepts a POST to change the logging level. In Spring Boot 1.5.x and 2.0.x, this works fine.

### Actual Behaviour
In Grails 3.x when using the `grails-logging` plugin, the log level change is registered (as verified through **/loggers**), but has no effect on the logging.

When the `grails-logging` dependency is commented out and the `@Slf4J` annotations are manually added to the Grails artifacts, the dynamic log level setting works fine.

### Example Application

This issue is thoroughly (some might say, excruciatingly) documented in the ReadMe at https://github.com/jfrosch/GrailsLoggingIssue.  The steps to reproduce it are included therein.

### Environment Information

- **Operating System**: Mac OSX and Windows 10
- **Grails Version:** 3.3.3
- **JDK Version:** Oracle JDK 1.8.0_161
- **Container Version (If Applicable):** N/A

### Example Application

- https://github.com/jfrosch/GrailsLoggingIssue

**Comment by niravassar** [comment]
Note this edit that should be made to the README.txt. with the text present it is confusing, but the img shows it should be striked from the doc and the testing makes more sense
![edit of readme](https://user-images.githubusercontent.com/18742260/42601258-a984badc-852a-11e8-9cb8-6c011fd64a04.png)

**Comment by jeffbrown** [comment]
@niravassar See the fork at https://github.com/jeffbrown/GrailsLoggingIssue.  That fork may be easier to follow.

**Comment by ColinHarrington** [comment]
I was able to reproduce the issue.  Working with Jeff's fork.  Looked like the trait wasn't required to reproduce the issue.  This is specific to Log4j2.  The Logback (default) implementation does work.

I got to the point where I was doing some pretty in-depth debugging trying to track down where the disconnect was. 

The `LoggersEndpoint`is the Actuator endpoint that they wanted to use to change log levels. Which delegates to the `LoggingSystem` https://github.com/spring-projects/spring-boot/blob/1.5.x/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/LoggersEndpoint.java#L92 For log4j that's the `Log4J2LoggingSystem` -> https://github.com/spring-projects/spring-boot/blob/1.5.x/spring-boot/src/main/java/org/springframework/boot/logging/log4j2/Log4J2LoggingSystem.java#L211

What I found is that the loggerConfig in the Log4j2LoggingSystem is disconnected from the logger that our `LoggingTransformer` in `grails-logging` creates.
